impl/src/planning/subplanners.rs

```diff
@@ -759,3 +777,52 @@ impl<'planner> CrateSubplanner<'planner> {
     }
   }
 }
+
+
+#[cfg(test)]
+mod tests {
+  use crate::testing::make_workspace;
+  use crate::metadata::CargoMetadataFetcher;
+  use crate::planning::{BuildPlannerImpl, BuildPlanner};
+  use std::path::PathBuf;
+  use crate::settings::{GenMode, RazeSettings};
+  use std::collections::HashMap;
+
+  #[test]
+  fn sha256s() {
+    let settings = RazeSettings {
+      genmode: GenMode::Remote,
+      index_url: "https://github.com/rust-lang/crates.io-index".to_string(),
+
+      workspace_path: String::default(),
+      incompatible_relative_workspace_path: bool::default(),
+      target: None,
+      targets: None,
+      binary_deps: HashMap::default(),
+      crates: HashMap::default(),
+      gen_workspace_prefix: String::default(),
+      output_buildfile_suffix: String::default(),
+      default_gen_buildrs: bool::default(),
+      registry: String::default(),
+    };
+
+    let cargo_toml = r#"[package]
+  name = "test"
+  version = "0.0.1"
+
+  [lib]
+  path = "not_a_file.rs"
+
+  [dependencies]
+  lazy_static = "=1.4.0"
+  "#;
+
+    let (_tempdir, files) = make_workspace(cargo_toml, None);
+
+    let mut fetcher = CargoMetadataFetcher::default();
+    let mut planner = BuildPlannerImpl::new(&mut fetcher);
+    let planned_build = planner.plan_build(&settings, &PathBuf::new(), files, None).unwrap();
```

Comment:
Any test in this file should be a direct test on a class in here. The fact that you're calling `planner.plan_build(...` indicates that this test should go into `planning.rs`